dartsim/src/WorldFeatures_TEST.cc

```diff
@@ -98,6 +123,76 @@ TEST_F(WorldFeaturesFixture, CollisionDetector)
   EXPECT_EQ("dart", world->GetCollisionDetector());
 }
 
+//////////////////////////////////////////////////
+TEST_F(WorldFeaturesFixture, Gravity)
+{
+  auto world = LoadWorld(this->engine, TEST_WORLD_DIR "/falling.world");
+  ASSERT_NE(nullptr, world);
+
+  // Check default gravity value
+  AssertVectorApprox vectorPredicate10(1e-10);
+  EXPECT_PRED_FORMAT2(vectorPredicate10, Eigen::Vector3d(0, 0, -9.8),
+                      world->GetGravity());
+
+  auto model = world->GetModel("sphere");
+  ASSERT_NE(nullptr, model);
+
+  auto link = model->GetLink(0);
+  ASSERT_NE(nullptr, link);
+
+  // initial link pose
+  const Eigen::Vector3d initialLinkPosition(0, 0, 2);
+  {
+    auto pos = link->FrameDataRelativeToWorld().pose.translation();
+    EXPECT_PRED_FORMAT2(vectorPredicate10,
+                        initialLinkPosition,
+                        pos);
+  }
+
+  auto linkFrameID = link->GetFrameID();
+
+  // Get default gravity in link frame, which is pitched by pi/4
+  EXPECT_PRED_FORMAT2(vectorPredicate10,
+                      Eigen::Vector3d(6.92964645563, 0, -6.92964645563),
+                      world->GetGravity(linkFrameID));
```

Comment:
somehow this call to `GetGravity` is causing an assertion in KinematicsFeatures.cc

https://github.com/ignitionrobotics/ign-physics/blob/e21c8c9f94d364a10f0671ec1eb347f2c32c3348/dartsim/src/KinematicsFeatures.cc#L33-L40

related to call to `detail::Resolve` in `detail/World.hh`:

https://github.com/ignitionrobotics/ign-physics/blob/ef1012d9170f962756d8dfa7668d6a43a099eea9/include/ignition/physics/detail/World.hh#L82-L100

I think we should relax the assertion, unless I'm going something wrong here

Reply:
I've disabled the assertion in 37d09ae

Reply:
I'm not sure if it hurts to ask the world frame what it's frame data is. Could be a problem for some physics engines? Looking at the code https://github.com/ignitionrobotics/ign-physics/blob/37d09ae717292623e6c1345efb26f66e458b0275/include/ignition/physics/detail/FrameSemantics.hh#L58-L62, maybe we should set `currentCoordinates = RotationType::Identity();`  if the `_relativeTo` is the world frame, as is done a couple of lines below.

Reply:
ooh, that's smart. I could leave the assert unchanged that way

Reply:
I've reverted the removal of the assert and followed your suggestion in 12b4cd442362efe76912558471d1f0d7e03d338e